### ISSUE TITLE
Allow sending a different project-root-uri during LSP initialize request

### ISSUE THREAD

**Issue by MTrost** [BetterThanTomorrow/calva#1866]
Calva's clojure-lsp does not work in a leiningen project where the project root and `project.clj` are in a subfolder:

```
- reporoot
  - code (=project root)
    - project.clj
```

When opening `reporoot` clojure-lsp does not load the correct classpath, when opening `code` everything works as expected. See also the discussion in this [Slack thread](https://clojurians.slack.com/archives/CBE668G4R/p1663413923486979).

@ericdallo pointed out the solution in Slack: "(...) the best would be if Calva allows users to send a different project-root-uri during LSP initialize request, lsp-mode (Emacs), for example allows that."

Minimal reproduction repo: https://github.com/MTrost/reporoot

**Comment by PEZ** [comment]
> the best would be if Calva allows users to send a different project-root-uri during LSP initialize request

I just tried this. (And got reminded that we've been trying it before.) VS Code won't allow it. Whatever I send, VS Code will use the workspace root. I think our options here are:

1. Send the project-root-uri in some non-standard way
2. Not support this use case

As bad as **2.** sounds, I've just found a way to trick VS Code to send the Leiningen project root here. It involves to use VS Code Workspace Folders. Like so:

0. With `reporoot` opened
1. **File -> Add Folder to Workspace...**
2. Select the `code` folder
3. Drag the folder roots (you now have two) so that `code` is before `reporoot`

VS Code will restart the clojure-lsp client and be using the Leiningen project root as desired. Then save this workspace and you won't have to repeat the four steps.
